Libvirt guests

petrkotas opened the pull request:
This pull requests add the deploy configuration for the libvirt-guests service.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ovirt/ovirt-ansible/62)
<!-- Reviewable:end -->

petrkotas committed:
Add readme for new libvirt-guests role

petrkotas committed:
Add default variables for libvirt-guests role

petrkotas committed:
Add galaxy meta info for libvirt-guests role

petrkotas committed:
Add VDSM version check

petrkotas committed:
Delete non needed variables

These variables are set by VDSM config itself.

petrkotas committed:
Change the variable type for parallel shutdown

petrkotas committed:
Add tasks to setup libvirt-guests

petrkotas committed:
Add dependency libvirt-guests to host-deploy

petrkotas committed:
Fix variable existence check

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,31 @@
+---
+- name: Fetch info about VDSM
+  command: bash -c "rpm -qi vdsm | grep -oE 'Version\\s+:\\s+[0-9\\.]+' | awk '{print $3}'"
```
1. See https://github.com/oVirt/ovirt-ansible/pull/56/files
2. I guess we should extract that VDSM version task to its own file, so it can be reused?

tremble replied:
This is probably better written as 
command: 'rpm -q --qf "%{version}" vdsm'

mwperina replied:
Ondro is there a way how to fetch VDSM version in ovirt-host-deploy role itself and afterwards store it in some variable and pass it to all subroles?

@machacekondra

tremble replied:
As long as it's in the same playbook (you can also do this as a role dependency) then variables set in one role will be available in another.

machacekondra replied:
Right. We can create something like hosts-facts role, which will be executed first.

petrkotas replied:
I have created the issue for this task so we do not forget about it. #70 
Does this patch need to wait for the #70 ? If not, can we move forward with the merge?
@machacekondra @mwperina @mykaul

tremble replied:
If you plan to move forward I would strongly urge you to replace the command with 

`  command: "rpm -q --qf '%{version}' vdsm"`

it's massively more readable than the grep+awk combination

machacekondra replied:
There is PR adding that new role:

 https://github.com/oVirt/ovirt-ansible/pull/71

Feel free to review.

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,31 @@
+    - name: Change default timeout if setup
+      lineinfile:
+        path: /etc/sysconfig/libvirt-guests
+        regexp: "^SHUTDOWN_TIMEOUT="
```
If it's an INI file, why not use the INI files module instead of regexp?

petrkotas replied:
Yaniv, thanks for pointing this out. ini_file is indeed better suited for this, checked it and it works the same and it is more readable.

petrkotas commented:
ci build please

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,31 @@
+      when:
+        - host_deploy_shutdown_parallel is defined
+
+    - name: Stop libvirt-guests service if running
```
stop? Why?

tremble replied:
Name says stop, definition says start...

mwperina replied:
Hmm, shouldn't the service be rather restarted? If it's started already before this role invocation and we will change the options, the service won't be restarted using this code ...

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/meta/main.yml:
```diff
@@ -0,0 +1,20 @@
+    - 7
+  - name: Fedora
+    versions:
+    - 25
```
What about 26? Why limit the platforms?

mwperina replied:
Ondro, do we need always to specify platforms for Galaxy? If so then we will need to include FC26 and remove FC24 for all roles in master branch ...
@machacekondra

petrkotas replied:
I will wait for the resolution.

machacekondra replied:
No really, but we dont support vdsm on anything different then fc and el, no? So I think it's better if it explicitly said.

michalskrivanek replied:
@machacekondra, better in what aspect? Does it help with anything?

mwperina replied:
So let's have there only CentOS 7, because our Fedora support will never be up-to-date

machacekondra replied:
It doesn't help with anything it's just a text generated in galaxy README, which I guess wouldn't be used by anyone, because this is just internal role.

machacekondra replied:
@petrkotas as Martin suggested, can you please remove Fedora from here?

machacekondra commented:
ci build please

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/README.md:
````diff
@@ -0,0 +1,48 @@
+----------------
+
+```yaml
+- name: oVirt host deploy - firewalld
````
copy-paste issue?

petrkotas replied:
As with previous.

machacekondra commented:
@mwperina

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/README.md:
```diff
@@ -0,0 +1,48 @@
+
+| Name                          | Default value  | Description                                                       |
+|-------------------------------|----------------|-------------------------------------------------------------------|
+| host_deploy_cluster_version   | 4.2            | The version of the cluster where the host resides. Based on this version the role will enable cluster version specific firewalld rules. |
```
firewalld? copy-paste issue?

petrkotas replied:
Hey Yaniv, yes copy & paste. Simple to solve.

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/README.md:
```diff
@@ -0,0 +1,48 @@
+To make the libvirt-guests play nicely with the VDSM, these options are set:
+ - the on-boot event is disabled
+ - the on-shutdown event is set to shutdown, instead of hibernate
+ - the default timeout is risen to 600 seconds and can be configured
```
risen -> raised

mykaul commented on roles/ovirt-host-deploy-libvirt-guests/defaults/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+host_deploy_firewalld_cluster_specific: []
+host_deploy_cluster_version: 4.2
```
Where is it used?

petrkotas replied:
Same with previous, copy&paste.

mwperina commented on roles/ovirt-host-deploy-libvirt-guests/defaults/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+host_deploy_firewalld_cluster_specific: []
```
Cut&paste?

michalskrivanek commented on roles/ovirt-host-deploy-libvirt-guests/README.md:
```diff
@@ -0,0 +1,48 @@
+  hosts: hostname
+
+  vars:
+    host_deploy_cluster_version: 4.2
```
since you're using vdsm package version check, I do not see how is cluster version of any more help

petrkotas committed:
Remove not needed variables

petrkotas committed:
Fix copy pase error

petrkotas committed:
Change use of lineinfile to ini_file module

machacekondra commented:
add to whitelist

petrkotas commented:
*[roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml, line 24 at r1](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62#-KsmC3HOP6j7LvtRTWQd-r1-24:-Ksnccx5DAamTn0e0Q3N:b-44d5ic) ([raw file](https://github.com/ovirt/ovirt-ansible/blob/bc73613bff34ee2d1a86cd2784118e2ce0308af3/roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml#L24)):*
<details><summary><i>Previously, mwperina (Martin Perina) wrote…</i></summary><blockquote>

Hmm, shouldn't the service be rather restarted? If it's started already before this role invocation and we will change the options, the service won't be restarted using this code ...
</blockquote></details>

For the libvirt-guests I need to ensure it is started. It does not need to be restarted. It is just a plain script, that loads the variables every time it runs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62)*
<!-- Sent from Reviewable.io -->

petrkotas commented:
*[roles/ovirt-host-deploy-libvirt-guests/README.md, line 37 at r1](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62#-KsmC3HOP6j7LvtRTWQa-r1-37:-KsncviN1_0369uVbUAv:b6n2wnk) ([raw file](https://github.com/ovirt/ovirt-ansible/blob/bc73613bff34ee2d1a86cd2784118e2ce0308af3/roles/ovirt-host-deploy-libvirt-guests/README.md#L37)):*
<details><summary><i>Previously, michalskrivanek wrote…</i></summary><blockquote>

since you're using vdsm package version check, I do not see how is cluster version of any more help
</blockquote></details>

It is just copy paster artifact. It is gone now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62)*
<!-- Sent from Reviewable.io -->

petrkotas commented:
*[roles/ovirt-host-deploy-libvirt-guests/defaults/main.yml, line 2 at r1](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62#-KsmC3HOP6j7LvtRTWQb-r1-2:-Ksnd00xm1AONUohSjNr:b6h2gq) ([raw file](https://github.com/ovirt/ovirt-ansible/blob/bc73613bff34ee2d1a86cd2784118e2ce0308af3/roles/ovirt-host-deploy-libvirt-guests/defaults/main.yml#L2)):*
<details><summary><i>Previously, mwperina (Martin Perina) wrote…</i></summary><blockquote>

Cut&paste?
</blockquote></details>

Indeed, it is gone now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62)*
<!-- Sent from Reviewable.io -->

petrkotas commented:
*[roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml, line 3 at r1](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62#-KsmC3HOP6j7LvtRTWQd-r1-3:-KsndDz8F4f_KNkCTaoy:bu3re1x) ([raw file](https://github.com/ovirt/ovirt-ansible/blob/bc73613bff34ee2d1a86cd2784118e2ce0308af3/roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml#L3)):*
<details><summary><i>Previously, tremble (Mark Chappell) wrote…</i></summary><blockquote>

As long as it's in the same playbook (you can also do this as a role dependency) then variables set in one role will be available in another.
</blockquote></details>

This means, that ovirt-host-deploy needs to do this and then every other role can profit. Is that right?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62)*
<!-- Sent from Reviewable.io -->

petrkotas committed:
Fix typo

petrkotas commented:
*[roles/ovirt-host-deploy-libvirt-guests/README.md, line 8 at r1](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62#-KsmC3HOP6j7LvtRTWQa-r1-8:-KsndsI-im0Znvpqtp6O:bw8ysw8) ([raw file](https://github.com/ovirt/ovirt-ansible/blob/bc73613bff34ee2d1a86cd2784118e2ce0308af3/roles/ovirt-host-deploy-libvirt-guests/README.md#L8)):*
<details><summary><i>Previously, mykaul (Yaniv Kaul) wrote…</i></summary><blockquote>

risen -> raised
</blockquote></details>

Fixed, thank you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ovirt/ovirt-ansible/62)*
<!-- Sent from Reviewable.io -->

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/meta/main.yml:
```diff
@@ -0,0 +1,20 @@
+
+  license: Apache License 2.0
+
+  min_ansible_version: 2.3
```
Here ans in README you refer to different min. Ansible versions.

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+        option: "SHUTDOWN_TIMEOUT"
+        value: "{{ host_deploy_shutdown_timeout }}"
+      when:
+        - host_deploy_shutdown_timeout is defined
```
So whatabout the defaults, you,ve mentioned in readme?

petrkotas replied:
With current deployment the defaults handling is tricky. The default values are given by controller from VDSM itself during VDSM deploy phase on the host.
This opens up a discussion for where to start putting the configuration, should it stay in the otopi plugin inside the VDSM or should it be completely handled by ansible?
@michalskrivanek  @mwperina what do you think?

michalskrivanek replied:
@petrkotas where is it in otopi plugins?

machacekondra replied:
Ok, then I think you should remove the defaults from the README, and say it don't have any default value. and also please remove the `default` directory.

petrkotas replied:
@michalskrivanek it is placed in configurators https://github.com/oVirt/vdsm/blob/master/lib/vdsm/tool/configurators/libvirt.py this is the libvirt one. The libvirt guests configuration can be seen in my pull request on gerrit https://gerrit.ovirt.org/#/c/79840/15/lib/vdsm/tool/configurators/libvirt.py

My opinion is, that once we fully transition towards the ansible. We can completely omit the old controllers and use ansible to fully control the host.

What do you think?

michalskrivanek replied:
then why bother with 79840?

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+        option: "PARALLEL_SHUTDOWN "
+        value: "{{ host_deploy_shutdown_parallel }}"
+      when:
+        - host_deploy_shutdown_parallel is defined
```
So whatabout the defaults, you,ve mentioned in readme?

petrkotas replied:
As already mentioned.

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/defaults/main.yml:
```diff
@@ -0,0 +1,2 @@
+---
```
Missing the default values for your variables here?

petrkotas replied:
As already mentioned.

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+        - host_deploy_shutdown_parallel is defined
+
+    - name: Start libvirt-guests service
+      service:
```
Is this restart save? Should we restart it only when configuration is changed?

petrkotas replied:
It is actually never supposed to restart. I only need to ensure it is started and enabled. That is all.
If the service is restarted it will perform is shutdown event and will shutdown all registered VMs. 

This behaviour is given by its implementation:
- the libvirt-guests is old init-v script ported to work in systemd, by very simple method of making it oneshot command executed on every registered event [startup, shutdown, restart, suspend]
- everytime libvirt-guests is fired-up it reloads the config file and performs its registered actions
- shutdown is trigered on systemd stop
- start is trigered on systemd start
- shutdown is triggered on systemd restart

michalskrivanek replied:
@petrkotas, so then you do not want to stop and start it while some VMs are running during this task, do you?

machacekondra replied:
But still if you change the configuration, shouldn't you restart the service? And if there are no changes just make sure service is running and don't restart it?

petrkotas replied:
@machacekondra you do not need to, that is the beauty of the old script. The configuration gets loaded every time the script gets executed, it does not run as daemon.

@michalskrivanek of course not, therefore you do not restart it and do not touch it until absolutely necessary. This is the reason, why you start it and do not restart.

petrkotas committed:
Remove version check and use known fact

petrkotas committed:
Rename service name to be more clear

petrkotas committed:
Fix the ansible version

petrkotas committed:
Removes default values from the role

petrkotas committed:
Change description to what it really does

petrkotas commented:
@machacekondra I removed the defaults and fixed your remarks. Thank You for pointing that out. I totally missed the wrong version in galaxy file. Can you please review? Thank you

ifireball commented:
ci test please

petrkotas committed:
Add default variables and setup of config file

petrkotas commented:
@machacekondra @michalskrivanek The configuration is moved to the ansible setup. It is now more readable and more clear for anybody interested in setup.
This way we can have more consistent deploy without fragmentation to multiple setup tools. 
PR

machacekondra commented on roles/ovirt-host-deploy-libvirt-guests/README.md:
```diff
@@ -0,0 +1,46 @@
+| Name                          | Default |  Description                                       |
+|-------------------------------|---------|----------------------------------------------------|
+| host_deploy_shutdown_timeout  |     600 | Timeout before systemd starts killing processes    |
+| host_deploy_shutdown_parallel |       0 | If set VMs are shutdown in parallel with count set |
```
Can you please document here `libvirt_guests_config` ?

petrkotas replied:
@machacekondra thank you for the input, it is done

petrkotas committed:
Add description for libvirt_guests_config

petrkotas committed:
Remove Fedora from galaxy file

machacekondra commented:
Looks good to me, thanks. @mwperina Martin can you remove the 'change requested', if it's OK by you? I will merge then.

petrkotas commented:
@mwperina Can you please take a look at this pull request? Is it OK with you?

mwperina reviewed (approved):
Looks good to me, but let's wait with merging until it passes OST successfully

machacekondra commented:
OST passed, I am merging.
